overcast/runner/__init__.py

```diff
@@ -373,7 +415,7 @@ def wait():
 
     def shell_step_cmd(self, details, env_prefix=''):
         if details.get('type', None) == 'remote':
-             node = self.nodes[details['node']][1]
+              node = next(iter(self.nodes[details['node']].fip_addresses))
```

Comment:
I would do for loop then, I think since the stack support a node can have multple fips, the code also support it.